src/app/preferences/general-preferences/general-preferences.service.ts

```diff
@@ -1,18 +1,21 @@
 namespace Core {
+  export enum state {
+    show = "show",
+    hide = "hide"
+  }
   export class GeneralPreferencesService {
-    private static readonly DEFAULT_VERTICAL_NAV_STATE = "show";
+    private static readonly DEFAULT_VERTICAL_NAV_STATE = state["show"];
     constructor(private $window: ng.IWindowService) {
       'ngInject';
     }
-    getDefaultVerticalNavState(): string {
-      if (this.$window.localStorage.getItem('defaultVerticalNavState') !== null) {
-        return this.$window.localStorage.getItem('defaultVerticalNavState');
+    getDefaultVerticalNavState(): state {
+      if (this.$window.localStorage.getItem('defaultVerticalNavState') != null) {
+        return (<any>state)[this.$window.localStorage.getItem('defaultVerticalNavState')];
```

Comment:
Why do you need to use array here?

And remember that type cast (`<any>...`) is the last thing to do in TypeScript as it's the compromise of the value of using TypeScript. Generally, you'd need to pay all the efforts to avoid it.